src/token/token.service.ts

```diff
@@ -138,12 +146,12 @@ export class TokenService {
   }
   async stakingRatio() {
     return (
-      (Number(await this.dposBalance()) /
+      (Number(await this.delegationService.totalDelegated()) /
         (Number(await this.totalSupply()) - Number(await this.totalLocked()))) *
       100
-    );
+    ).toString();
   }
-  async marketDetails() {
+  async marketDetails(asString?: boolean) {
```

Comment:
This flag shouldn't be here. It will break the cached version 'cause you are not taking that into account